File: 2324A/NewBB mocks/mock2/integerlists_Amanda_indexes.py
'''
OpenKattis
integerlists
Code by: Amanda Lim
'''

T = int(input())
for t in range(T):
    BAPC = input().strip()

    N = int(input())
    list_of_int = input().strip()
    list_of_int = list_of_int[1:-1]
    
    list_of_int = list_of_int.split(',')
    if (N == 0):
        list_of_int.clear()
    
    A = 0
    B = N
    reverse = False
    error = False
    for i in BAPC:
        if i == 'R':
            reverse = not reverse
        else:
            if A >= B:
                error = True
                break
            else:
                if reverse:
                    B -= 1
                else:
                    A += 1
    
    if error:
        print("error")
    else:
        print('[', end="")
        if reverse:
            for i in range(B - 1, A - 1,-1):
                print(list_of_int[i], end="")
                if (i > A):
                    print(',', end="")
        else:
            for i in range(A,B):
                print(list_of_int[i], end="")
                if (i < B - 1):
                    print(',', end="")
        print(']')

File: 2324A/NewBB mocks/mock2/integerlists_Amanda_indexes.cpp
/*
*OpenKattis
*integerlists
*Code by: Amanda Lim
*/

#include <bits/stdc++.h>
using namespace std;

int main ()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    
    int T;
    cin >> T;
    for (int t = 0; t < T; t++) {
        string BAPC;
        cin >> BAPC;

        int N;
        cin >> N;

        char tapon;
        cin >> tapon;

        int list_of_int[N];
        for (int n = 0; n < N; n++)
            cin >> list_of_int[n] >> tapon;
        if (!N)
            cin >> tapon;

        bool error = false;
        bool reverse = false;
        int A = 0, B = N;
        for (int i = 0; i < (int) BAPC.size(); i++) {
            if (BAPC[i] == 'R')
                reverse = !reverse;
            else {
                if (A >= B) {
                    error = true;
                    break;
                }

                if (!reverse)
                    A++;
                else
                    B--;
            }
        }

        if (error)
            cout << "error" << endl;
        else {
            cout << '[';
            if (!reverse)
                for (int i = A; i < B; i++) {
                    cout << list_of_int[i];
                    if (i < (B - 1))
                        cout << ',';
                }
            else
                for (int i = B - 1; i >= A; i--) {
                    cout << list_of_int[i];
                    if (i > A)
                        cout << ',';
                }
            cout << ']' << endl;
        }
    }

    return 0;
}